theano/tensor/extra_ops.py

```diff
@@ -511,6 +511,29 @@ def squeeze(x):
     return view
 
 
+def compress(condition, x, axis=None, out=None):
+    """Return selected slices of an array along given axis.
+
+    It returns the input array, but with the
+    broadcastable dimensions removed. This is
+    always `x` itself or a view into `x`.
+    Wrapping of numpy.compress
+
+    :param x: 1 dimension, bools
+
+    :param condition: array of the same shape as x with corresponding weights.
+        Optional.
+
+    :return: `x` with selected slices
+
+    .. versionadded:: 0.7
+    """
+    # This is done to keep the same function signature then NumPy.
+    assert out is None
```

Comment:
Currently, we do not add extra parameter at the end for numpy compatibility. But we do it in the middle, to keep the order the same.

I understand why we could add it for better compatibility, but i think having only this case support is worst then not supporting it. People will try to expect that we support it elsewhere and we don't. So I would refer the out parameter is removed from this PR.

If someone think it is important enought, I think we should discuss the use case on the mailing list and add support for it to many function at the same time if we support it. But most numpy code do not use the out parameter, so it won't affect much code.